File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int res = 0;

        for (int num: nums) {
            set.add(num);
        }

        for (int i = 0; i < nums.length; i++) {
            int curr = nums[i];
            int streakLength = 0;

            while (set.contains(curr)) {
                streakLength++;
                curr+=1;
            }

        res = Math.max(res, streakLength);
        }
        return res;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-2.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res = nums[0]
        currMax = currMin = 1

        for num in nums:
            tmp = currMax * num
            currMax = max(num, num * currMax, currMin * num)
            currMin = min(num, currMin * num, tmp)
            res = max(currMax, res)

        return res
        

# 2 -> max 2 min 1
# 4 -> max 8 min